contributing/topics/guide-breaking-changes.md

```diff
@@ -1,6 +1,252 @@
+            MySqlFlexibleServerResource{},
+        }
+        
+        if !features.FivePointOhBeta() {
+            resources = append(resources, ExampleResource{})
+        }
+        
+        return resources
+    }
+    ```
+    
+    For Untyped Resources
+    ```go
+    func (r Registration) SupportedResources() map[string]*pluginsdk.Resource {
+        resources := map[string]*pluginsdk.Resource{
+            "azurerm_mysql_flexible_server": resourceMysqlFlexibleServer(),
+    
+        }
+    
+        if !features.FivePointOhBeta() {
+            resources["azurerm_example"] = resourceExample()
+        }
+    
+        return resources
+    }
+    ```
+
+3. Skip all tests related to the deprecated resource.
+
+    ```go
+    func TestAccExample_basic(t *testing.T) {
+        if features.FivePointOhBeta() {
+            t.Skipf("Skipping since `azurerm_example` is deprecated and will be removed in 5.0")
+        }
+        data := acceptance.BuildTestData(t, "azurerm_example", "test")
+        r := ExampleResource{}
+        
+        data.ResourceTest(t, r, []acceptance.TestStep{
+        {
+            Config: r.basic(data),
+            Check: acceptance.ComposeTestCheckFunc(
+                check.That(data.ResourceName).ExistsInAzure(r),
+            ),
+        },
+            data.ImportStep()
+        })
+    }
+    ```
+
+4. Update the upgrade guide under `website/docs/5.0-upgrade-guide.markdown`.
+
+   ```markdown
+   ## Removed Resources
+   
+   ### `azurerm_example`
+   
+   This deprecated resources has been removed from the Azure Provider.
+   ```
+   **Note:** The Resource documentation does not need to be updated with a note since the provider will inform any users using the resource of the deprecation with a warning.
+
+
+## Breaking Schema Changes and Deprecations
+
+Breaking schema changes can include:
+- Property renames
+- When properties become Required
+- When properties have Computed removed in some cases
+- Changes to the validation e.g. the validation becomes more restrictive
+- Changing the default value
+
+In all cases the deprecation is handled the same way and will be illustrated by the example below.
+
+The following example follows a fictional resource that will have the following breaking changes made:
+- The property `enable_scaling` renamed to `scaling_enabled`
+- The property `version` has its default changed from `1` to `2`
+
+1. Update the Schema with the target or desired breaking schema change and patch over the breaking schema change with the current behaviour using the major release feature flag.
+
+   ```go
+   func (r ExampleResource) Arguments() map[string]*pluginsdk.Schema{
+      args := map[string]*pluginsdk.Schema{
+         "scaling_enabled": {
+            Type:     pluginsdk.TypeBool,
+            Optional: true,
+            Default: false,
+         },      
+         "version": {
+            Type:     pluginsdk.TypeString,
+            Optional: true,
+            Default: 1,
```

Comment:
Minor but I believe this wants to be 2 and the Default below wants to be 1 based on the wording of the example default going from 1 to 2